src/pages/dataTableUtilities/actions/tableActions.js

```diff
@@ -279,3 +280,30 @@ export const TableActionsLookup = {
   addTransactionItem,
   addStocktakeBatch,
 };
+
+/**
+ * =====================================================================
+ *
+ *                             Overrides
+ *
+ * Below are actions which are overrides of base actions.
+ *
+ * Example: editCountedQuantityWithReason overrides editCountedQuantity
+ * for a stocktakeEditPage when reasons are defined.
+ *
+ * =====================================================================
+ */
+
+export const refreshDataWithFinalisedToggle = () => ({
```

Comment:
This is probably more than you wanted but the two questions are related (the one above) and are questions I've been struggling with. This question boils down to a pretty HOT TOPIC - Fat reducer v fat action creators. Putting my thoughts here so I can rationalise my position on it and get your thoughts.

There is no 'right' or 'more flux' way to do this, it's still up in the air which is better.

Quote from redux documentation on the topic:

`Find the balance between these two extremes, and you will master Redux.`

So there's two techniques for handling conditionals that i've used, essentially.

- Having basic IFs in an action creator
- 'Overriding' actions

#### Having basic conditions in an action creator
- One big advantage to using flux is that actions are very understandable and simple. Having explicit actions for ON/OFF, YES/NO rather than a single action costs almost nothing. 
- The relationship between action and reducer is many to many. If a reducer wants to act on an action but requires state, if it's not the right reducer, it can't as it can't access the state.
- It's alot easier for middleware to help if the action is 'self-contained'. I.e. consider a queue of the last 100 actions. That would be a lot less useful if the action is "TOGGLE_X" rather than "TOGGLE_X_OFF"
- Keeps the responsibilities minimal. With a thin action creator, the reducer determines what should happen given an action and also calculates the new state. Rather than action creator determines the action, reducer calculates the state for that action.

I have opted to put all (i think, at least) conditional logic in action creators. This, in my opinion, gives more control over the flow of the application. Action creator is first in the lifecycle of a dispatched action, whereas the reducer is last. Conditional logic then can therefore not be used for side effects, if in the reducer.

#### 'Overriding' actions
An alternative to this would have in an action creator `IF X: return thisAction()` etc. OR in the reducer `IF X: do this, else do something different else this thing`

The main purpose for this is for the open-closed principle and to try to use the strategy pattern. Dynamically determining the algorithm at runtime. By determining the page actions at runtime, we can very easily completely change logic of any page given any sort of customization, without touching the UI or any previously written code. If we did this just in a action creator, it would be a string of IF's/switch which violates the open-closed principle